gnosis/eth/tests/eip712/test_eip712.py

```diff
@@ -33,59 +68,70 @@ def test_eip712_encode_hash(self):
-                {"name": "name", "type": "string"},
-                {"name": "version", "type": "string"},
-                {"name": "chainId", "type": "uint256"},
-                {"name": "verifyingContract", "type": "address"},
-            ],
-            "Mailbox": [
-                {"name": "owner", "type": "address"},
-                {"name": "messages", "type": "Message[]"},
-            ],
-            "Message": [
-                {"name": "sender", "type": "address"},
-                {"name": "subject", "type": "string"},
-                {"name": "isSpam", "type": "bool"},
-                {"name": "body", "type": "string"},
-            ],
-        }
-
-        msgs = [
-            {
-                "sender": address,
-                "subject": "Hello World",
-                "body": "The sparrow flies at midnight.",
-                "isSpam": False,
-            },
-            {
-                "sender": address,
-                "subject": "You may have already Won! :dumb-emoji:",
-                "body": "Click here for sweepstakes!",
-                "isSpam": True,
-            },
-        ]
-
-        mailbox = {"owner": address, "messages": msgs}
-
         payload = {
             "types": wrong_types,
             "primaryType": "Mailbox",
             "domain": {
                 "name": "MyDApp",
                 "version": "3.0",
                 "chainId": 41,
-                "verifyingContract": address,
+                "verifyingContract": self.address,
             },
-            "message": mailbox,
+            "message": self.mailbox,
         }
 
         with self.assertRaises(ValueError):
             eip712_encode_hash(payload)
 
-        payload["types"] = types
+        payload["types"] = self.types
         self.assertEqual(
             eip712_encode_hash(payload),
             b"\xd5N\xcbf7\xfa\x99\n\xae\x02\x86\xd4 \xacpe\x8d\xb9\x95\xaem\t\xcc\x9b\xb1\xda\xcf6J\x14\x17\xd0",
         )
+
+    def test_eip712_encode_hash_string_uint(self):
+        # test string uint (chainId)
+        payload = {
+            "types": self.types,
+            "primaryType": "Mailbox",
+            "domain": {
+                "name": "MyDApp",
+                "version": "3.0",
+                "chainId": "41",
+                "verifyingContract": self.address,
+            },
+            "message": self.mailbox,
+        }
+
+        self.assertEqual(
+            eip712_encode_hash(payload),
+            b"\xd5N\xcbf7\xfa\x99\n\xae\x02\x86\xd4 \xacpe\x8d\xb9\x95\xaem\t\xcc\x9b\xb1\xda\xcf6J\x14\x17\xd0",
+        )
+
+    def test_eip712_encode_hash_string_bytes(self):
+        types_with_bytes = {
+            "EIP712Domain": [
+                {"name": "name", "type": "string"},
+            ],
+            "Message": [
+                {"name": "oneByte", "type": "bytes1"},
+                {"name": "maxByte", "type": "bytes32"},
+            ],
+        }
+        # test string uint (chainId)
```

Comment:
nitpick: this comment does not apply to this test right?

Reply:
Yes, copy pasted from above, done https://github.com/safe-global/safe-eth-py/pull/427/commits/fa4a21f60b881a33fd2c0b8a72aa3c5c3c74b188